root/scripts/createRecipe.js

```diff
@@ -7,64 +7,98 @@ import {
 
 // const createRecipe = document.querySelector(document.getElementById('Create'));
 // const deleteRecipe = document.querySelector(document.getElementById('Delete'));
-let i = 6; // instructions counter
+let i = 1; // instructions counter
 let ingCount = 1; // Ingredient Counter
 
 function addStep() {
   const instructions = document.querySelector('.instructions');
   const steps = document.createElement('input');
-  steps.setAttribute('class', 'step');
-  steps.setAttribute('type', 'text');
-  steps.setAttribute('placeholder', `Step ${i.toString()}`);
-  steps.setAttribute('id', `Step${i.toString()}`);
-  instructions.appendChild(steps);
-  i += 1;
+  let blank = false;
+  for (let j = 1; j < i; j += 1) {
+    const lastid = `Step${j}`;
+    const lastinput = document.getElementById(lastid).value;
+    if (lastinput.length === 0) {
+      blank = true;
+      break;
+    }
+  }
+  if (blank) {
+    alert('blank inputs are not allowed!');
+  } else {
+    steps.setAttribute('class', 'step');
+    steps.setAttribute('type', 'text');
+    steps.setAttribute('placeholder', `Step ${i.toString()}`);
+    steps.setAttribute('id', `Step${i.toString()}`);
+    instructions.appendChild(steps);
+    i += 1;
+  }
 }
 
 function deleteStep() {
   i -= 1;
+  /*
   if (i < 2) {
     i = 2;
   }
+  */
   const stepStr = `Step${i.toString()}`;
   const lastStep = document.getElementById(stepStr);
   lastStep.remove();
 }
 
 function addIng() {
-  const ingredSteps = document.querySelector('.ingredSteps');
-  const amount = document.createElement('input');
-  amount.setAttribute('type', 'text');
-  amount.setAttribute('class', 'Ingre');
-  amount.setAttribute('placeholder', 'Amount');
-  amount.setAttribute('id', `amount${ingCount.toString()}`);
-
-  const units = document.createElement('input');
-  units.setAttribute('type', 'text');
-  units.setAttribute('class', 'unit');
-  units.setAttribute('placeholder', 'Units');
-  units.setAttribute('id', `units${ingCount.toString()}`);
-
-  const steps = document.createElement('input');
-  steps.setAttribute('type', 'text');
-  steps.setAttribute('class', 'Ingredient');
-  steps.setAttribute('placeholder', `Ingredient ${ingCount.toString()}`);
-  steps.setAttribute('id', `ing${ingCount.toString()}`);
-
-  // const breakline = document.createElement('br');
-
-  ingredSteps.appendChild(amount);
-  ingredSteps.appendChild(units);
-  ingredSteps.appendChild(steps);
-  // ingredSteps.appendChild(breakline);
-  ingCount += 1;
+  let blank = false;
+  for (let j = 1; j < ingCount; j += 1) {
+    const lastid1 = `amount${j}`;
+    //const lastid2 = `units${j}`;
+    const lastid3 = `ing${j}`;
+    const lastinput1 = document.getElementById(lastid1).value;
+    //const lastinput2 = document.getElementById(lastid2).value;
```

Comment:
Codacy has a fix for the issue: [Expected exception block, space or tab after '//' in comment.](https://app.codacy.com/gh/cse110-fa21-group34/rocketrecipes/pullRequest?prid=8480856)
```suggestion
    // const lastinput2 = document.getElementById(lastid2).value;
```